composer/datasets/in_context_learning_evaluation.py

```diff
@@ -1288,15 +1294,26 @@ def __init__(
             'generation_kwargs': {
                 'pad_token_id': self.pad_tok_id,
                 'num_beams': 1,  # single beam
-                'num_return_sequences': generations_per_sample,
                 'do_sample': True,
+                'temperature': 0.2, # good default for code
                 'use_cache': True,
                 'eos_token_id': self.tokenizer.eos_token_id
-            }
+            },
+            'sample_id': [],
+            'pass_at_k': list(pass_at_k),
+            'generations_per_sample': generations_per_sample,
+            'dataset_size': dataset_size,
         }
         if 'generation_kwargs' in kwargs:
             self.update_generation_kwargs(kwargs['generation_kwargs'])
 
+    def repeat_dataset(self, dataset: HFDataset, repetitions: int) -> HFDataset:
+        df = dataset.to_pandas()
```

Comment:
I think this is a little bit dangerous because it puts the whole dataset in memory, something that is otherwise avoided because everything else uses arrow datasets. Is there an alternative approach that doesn't required materializing N copies of the dataset in memory.

Reply:
Sounds good, refactored with Dataset.from_generator